Automattic/jetpack
# SSO: disable WPCOM invitation functionality for non-connected users

## Proposed changes:
PR #35902 introduced the new "Invite user to WordPress.com" checkbox that queries WPCOM API via user token.
The checkbox is displayed for all users, but those non-connected don't have a user token, so they aren't able to add new WordPress user accounts with the checkbox checked.

- Hide the "Invite user to WordPress.com" checkbox for non-connected users.
- Disable the invitation sending functionality for non-connected users.

### Other information:


## Jetpack product discussion
p1711145757980779-slack-C06QSU59PHS

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Fully connect a Jetpack site.
2. Go to "Users -> Add New User". Confirm you see the "Invite user to WordPress.com" checkbox and it's checked by default. Keep it checked.
3. Create the user with a valid email, confirm you receive an invitation to that email.
4. Login under a non-connected user, go to "Users -> Add New User". Confirm you don't see the "Invite user to WordPress.com" checkbox.
